base_layer/wallet/src/base_node_service/service.rs

```diff
@@ -389,8 +392,12 @@ where
                 Ok(BaseNodeServiceResponse::BaseNodePeerSet)
             },
             BaseNodeServiceRequest::GetChainMetadata => match self.state.chain_metadata.clone() {
-                Some(v) => Ok(BaseNodeServiceResponse::ChainMetadata(Some(v))),
-                None => Ok(BaseNodeServiceResponse::ChainMetadata(self.db.get_chain_meta().await?)),
+                Some(metadata) => Ok(BaseNodeServiceResponse::ChainMetadata(Some(metadata))),
+                None => {
+                    // if we don't have live state, check if we've previously stored state in the wallet db
+                    let metadata = self.db.get_chain_metadata().await?;
```

Comment:
Should maybe cater for the case that neither of these case are true e.g. in a completed fresh wallet before the first chain meta data response is received.

Reply:
yeah so `get_chain_metadata` now returns an option so that would be `None`